File: api/src/main/java/api/security/service/JwtTokenService.java
package api.security.service;

import api.entity.Role;
import api.security.model.JwtUserDetails;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Set;

public interface JwtTokenService {

    void updateRefreshToken(String username, String refreshToken);
    String generateAccessToken(UserDetails userDetails);
    String generateRefreshToken(UserDetails userDetails);
    String getUsernameFromJWT(String token);
    long getExpTimeFromJWT(String token);

    Set<Role> getUserRolesFromJWT(String token);

    JwtUserDetails getUserDetailsFromJWT(String token);

    boolean verifyToken(String authToken);
}

File: api/src/main/java/api/security/service/JwtTokenService.kt
package api.security.service

import api.entity.Role
import api.security.model.JwtUserDetails
import org.springframework.security.core.userdetails.UserDetails

interface JwtTokenService {

    fun updateRefreshToken(username: String, refreshToken: String)
    fun generateAccessToken(userDetails: UserDetails): String
    fun generateRefreshToken(userDetails: UserDetails): String
    fun getUsernameFromJWT(token: String): String
    fun getExpTimeFromJWT(token: String): Long

    fun getUserRolesFromJWT(token: String): Set<Role>

    fun getUserDetailsFromJWT(token: String): JwtUserDetails

    fun verifyToken(authToken: String): Boolean
}
